docs/dev/table/tableApi.md

```diff
@@ -1682,6 +1682,34 @@ The `OverWindow` defines a range of rows over which aggregates are computed. `Ov
 
 {% top %}
 
+### Map
+
+Map performs a map operation with an user-defined function. The output will be flattened
+if the output type is composite type.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+ScalarFunction func = new MyMapFunction();
+tableEnv.registerFunction("func", func);
+
+Table table = input
+  .map(func("c")).as("a, b")
+{% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+val func = new MyMapFunction()
```

Comment:
val func = new MyMapFunction() => val func: ScalarFunction = new MyMapFunction() ？
To tell user that func is a `ScalarFunction`.